aws/retry.go

```diff
@@ -0,0 +1,78 @@
+package aws
```

Comment:
I suggest moving this to the `aws/internal/tfresource` package as that is where much functionality layered on the `github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource` package lives.
Plus it's one less source file in the `aws` package 😄.

Reply:
Plus, is this unit testable?

Reply:
Good call. I've moved it and add a unit test. Not a very good one but a place holder for future, better tests.